src/rust/engine/testutil/mock/src/execution_server.rs

```diff
@@ -169,7 +190,17 @@ impl MockResponder {
       .unwrap()
       .pop_front()
     {
-      Some((op, duration)) => {
+      Some(MockOperation { op: None, duration }) => {
```

Comment:
I think this would probably be clearer by splitting into two nested matches, rather than having everything be one top-level match
```rust
match ... {
  Some(mock_operation) {
    let MockOperation { op: maybe_op, duration: maybe_duration } = mock_operation;
    if let Some(duration) = maybe_duration {
      sleep(d);
    }
    if let Some(op) = maybe_op {
      // send
    } else {
      drop(sink);
    }
  },
  None => // fail
}
```